### PR TITLE
Add new electronics page

### PR DESCRIPTION
**Pull request by JosephBushagour**
Add a new electronics page with interactive JavaScript. Will also add 'Electronics' to the sidebar which will cause a small, resolvable merge conflict with the homepage.

Image of page below:
Electronics_Page

### CODE REVIEW

File: portfolio/src/main/webapp/electronics_info.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+// limitations under the License.
+
+// The info for each clickable item in the picture
+const info = {
+  tackleBox: 'A cheap fishing tackle-box I bought from a hardware store.\
+      They are the ideal way to organize your small components.',
+  ti81: 'A Ti-81 calculator I found at a thrift store. The LCD screen\
+      was broken when I found it, but upon further inspection some connections\
+      just required some re-soldering.',
+  computer: 'An old Dell Lattitude I found in a business liquidation auction.\
```
Comment (SarahGB) [inline portfolio/src/main/webapp/electronics_info.js:22]: Nit: Dell Latitude

File: portfolio/src/main/webapp/electronics_info.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      generic, hardware store flux as well as lead-free solder.',
+  tape: 'A mountain of tape for various purposes, including, but not limited\
+      to, electrical isolation, heat resistant adhesion, high voltage \
+      warning, and more.',
+  cannedAir: 'Canned air, when turned upside down, can be used as a cheap \
+      \"freezer spray\" which can help test for temperature instability in \
+      projects.',
+  snips: 'Probably the highest quality tool I own is this pair of Hakko flush\
+      cutters. They\'re a joy to use and are quite durable.',
+  LEDsTransistors: 'A box shared between LEDs (5mm and 3mm) as well as \
```
Comment (SarahGB) [inline portfolio/src/main/webapp/electronics_info.js:53]: For camelCase formatting, even initialisms are treated as lowercase. Please update this to ledsTransistors.

File: portfolio/src/main/webapp/electronics_info.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The info for each clickable item in the picture
+const info = {
+  tackleBox: 'A cheap fishing tackle-box I bought from a hardware store.\
+      They are the ideal way to organize your small components.',
+  ti81: 'A Ti-81 calculator I found at a thrift store. The LCD screen\
```
Comment (SarahGB) [inline portfolio/src/main/webapp/electronics_info.js:19]: Nit: I believe this is TI-81 (capital I)

File: portfolio/src/main/webapp/electronics_info.js (hunk 1 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/electronics_info.js]: Great catch! Fixed.

File: portfolio/src/main/webapp/electronics_info.js (hunk 2 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/electronics_info.js]: Great to know! Fix pushed.

File: portfolio/src/main/webapp/electronics_info.js (hunk 3 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/electronics_info.js]: You're absolutely correct! Fix pushed.

File: portfolio/src/main/webapp/electronics.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            shape="poly">
+        <area alt="Post-it Notes" onclick="moreInfo('postItNotes')" 
+            coords="350, 82, 350, 120, 371, 118, 369, 84" 
+            shape="poly">
+        <area alt="Halogen Light" onclick="moreInfo('light')" 
+            coords="19, 177, 25, 208, 33, 211, 53, 204, 67, 184, 59, 163, 37, 162" 
+            shape="poly">
+      </map>
+      <div id="electronics-info">
+        <p>Click on an item in the image for more information here.</p>
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/electronics.html:84]: Do you need both a <div> and a <p> here?

File: portfolio/src/main/webapp/electronics.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      </div>
+      <p>If you want to learn more about my electronics, or just electronics 
+        in general, check out the projects on my projects page.
+      </p>
+    </div>
+  </body>
+</html>
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/electronics.html:91]: nit: Add a newline at the end of the file (I'm not sure about Cloud Shell specifically but most editors have an option to just do that automatically)

File: portfolio/src/main/webapp/electronics_info.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The info for each clickable item in the picture
+const info = {
+  tackleBox: 'A cheap fishing tackle-box I bought from a hardware store.\
+      They are the ideal way to organize your small components.',
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/electronics_info.js:18]: Doesn't this end up adding a bunch of spaces in the middle of the text?

I would recommend 'Line 1' + 'Line 2' instead

File: portfolio/src/main/webapp/electronics_info.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+      generic, hardware store flux as well as lead-free solder.',
+  tape: 'A mountain of tape for various purposes, including, but not limited\
+      to, electrical isolation, heat resistant adhesion, high voltage \
+      warning, and more.',
+  cannedAir: 'Canned air, when turned upside down, can be used as a cheap \
+      \"freezer spray\" which can help test for temperature instability in \
+      projects.',
+  snips: 'Probably the highest quality tool I own is this pair of Hakko flush\
+      cutters. They\'re a joy to use and are quite durable.',
+  LedsTransistors: 'A box shared between LEDs (5mm and 3mm) as well as \
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/electronics_info.js:53]: nit: lowercase

File: portfolio/src/main/webapp/electronics_info.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+  postItNotes: 'Post-its are, in my opinion, the ideal way to leave little \
+      notes for future you. Plus, you can color-code them!',
+  light: 'A halogen light, useful not only for the photons, but the produced \
+      heat can help move solder fumes out of the way.',
+}
+
+/**
+ * Upon clicking an image, this function loads that description to the html.
+ */
+function moreInfo(subject) {
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/electronics_info.js:71]: nit: Make this function name more descriptive of what it actually does, eg. "displayMoreInfo"

File: portfolio/src/main/webapp/electronics_info.js (hunk 6 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/electronics_info.js]: Text inside the <p> tag will have extra spaces removed, which was why I had the JavaScript put this text in a <p> tag. I agree though, your way is less prone to error, and I have fixed it in code.

File: portfolio/src/main/webapp/electronics.html (hunk 4 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/electronics.html]: Good catch! Fix pushed.

File: portfolio/src/main/webapp/electronics_info.js (hunk 7 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/electronics_info.js]: Good catch! Fixed.

File: portfolio/src/main/webapp/electronics_info.js (hunk 8 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/electronics_info.js]: That's a very good point, fix pushed!

**Pull request merged**